apps/web/src/features/upload/utils.ts

```diff
@@ -401,41 +428,52 @@ function generateSampleData({
 }
 
 export function createUploadTemplateCSV(instrument: AnyUnilingualFormInstrument) {
-  // TODO - type validationSchema as object
-  const instrumentSchema = instrument.validationSchema as z.AnyZodObject;
+  try {
+    const instrumentSchema = instrument.validationSchema;
 
-  const instrumentSchemaDef: unknown = instrument.validationSchema._def;
+    if (!isZodObject(instrumentSchema)) {
+      throw new Error('Validation schema for this instrument is invalid');
+    }
 
-  let shape: { [key: string]: z.ZodTypeAny } = {};
+    const instrumentSchemaDef: unknown = instrument.validationSchema._def;
 
-  if (isZodTypeDef(instrumentSchemaDef) && isZodEffectsDef(instrumentSchemaDef)) {
-    const innerSchema: unknown = instrumentSchemaDef.schema._def;
-    if (isZodTypeDef(innerSchema) && isZodObjectDef(innerSchema)) {
-      shape = innerSchema.shape() as { [key: string]: z.ZodTypeAny };
+    let shape: { [key: string]: z.ZodTypeAny } = {};
+
+    if (isZodTypeDef(instrumentSchemaDef) && isZodEffectsDef(instrumentSchemaDef)) {
+      const innerSchemaDef: unknown = instrumentSchemaDef.schema._def;
+      if (isZodTypeDef(innerSchemaDef) && isZodObjectDef(innerSchemaDef)) {
+        shape = innerSchemaDef.shape() as { [key: string]: z.ZodTypeAny };
+      }
+    } else {
+      shape = instrumentSchema.shape as { [key: string]: z.ZodTypeAny };
     }
-  } else {
-    shape = instrumentSchema.shape as { [key: string]: z.ZodTypeAny };
-  }
 
-  const columnNames = Object.keys(shape);
+    const columnNames = Object.keys(shape);
 
-  const csvColumns = INTERNAL_HEADERS.concat(columnNames);
+    const csvColumns = INTERNAL_HEADERS.concat(columnNames);
 
-  const sampleData = [...INTERNAL_HEADERS_SAMPLE_DATA];
-  for (const col of columnNames) {
-    const typeNameResult = getZodTypeName(shape[col]!);
-    if (!typeNameResult.success) {
-      throw new Error(typeNameResult.message);
+    const sampleData = [...INTERNAL_HEADERS_SAMPLE_DATA];
+    for (const col of columnNames) {
+      const typeNameResult = getZodTypeName(shape[col]!);
+      if (!typeNameResult.success) {
+        throw new Error(typeNameResult.message);
+      }
+      sampleData.push(generateSampleData(typeNameResult));
     }
-    sampleData.push(generateSampleData(typeNameResult));
-  }
 
-  unparse([csvColumns, sampleData]);
+    unparse([csvColumns, sampleData]);
 
-  return {
-    content: unparse([csvColumns, sampleData]),
-    fileName: `${instrument.internal.name}_${instrument.internal.edition}_template.csv`
-  };
+    return {
+      content: unparse([csvColumns, sampleData]),
+      fileName: `${instrument.internal.name}_${instrument.internal.edition}_template.csv`
+    };
+  } catch (e: unknown) {
+    if (e instanceof Error && e.message) {
+      throw e;
+    }
+
+    throw new Error('Error generating Sample CSV template');
```

Comment:
Set `Error.cause` to be `e` in this case:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause